tasks/build-algolia-search.mjs

```diff
@@ -0,0 +1,228 @@
+  'react-native',
+  'angular',
+  'vue',
+  'ionic',
+  'next'
+];
+
+const pagesToSkip = ['/', '/ChooseFilterPage', '/404'];
+const pagesWithIndex = ['/cli/function', '/cli', '/console'];
+
+const pageValues = [];
+Object.keys(pathmap).forEach(async (key) => {
+  const value = pathmap[key];
+  let { page } = value;
+
+  if (pagesToSkip.includes(page)) {
+    return;
+  }
+
+  if (pagesWithIndex.includes(page)) {
+    page += '/index';
+  }
+
+  const filename = path.join(__dirname, '../src/pages', page) + '.mdx';
+  let platform = '';
+  if (page.includes('[')) {
+    platform = key.split('/').pop();
+  }
+  pageValues.push({ filename, platform });
+});
+
+try {
+  dotenv.config();
+
+  if (!process.env.PUBLIC_ALGOLIA_APP_ID) {
+    throw new Error('PUBLIC_ALGOLIA_APP_ID is not defined');
+  }
+
+  if (!process.env.ALGOLIA_SEARCH_ADMIN_KEY) {
+    throw new Error('ALGOLIA_SEARCH_ADMIN_KEY is not defined');
+  }
+
+  console.log('Compiling index...');
+
+  readPages(async function() {
+    const transformed = transformPostsToSearchObjects(articles);
+    const client = algoliasearch(
+      process.env.PUBLIC_ALGOLIA_APP_ID,
+      process.env.ALGOLIA_SEARCH_ADMIN_KEY
+    );
+    const index = client.initIndex('custom_search_staging');
+    const algoliaResponse = await index.saveObjects(transformed);
+
+    console.log(
+      `Successfully added ${algoliaResponse.objectIDs.length} records to Algolia search!`
+    );
+  });
+} catch (error) {
+  console.error(error);
+  process.exit(1);
+}
+
+function readPages(cb) {
+  if (!pageValues.length) {
+    if (cb) cb();
+    return;
+  }
+  const page = pageValues.pop();
+  const { filename, platform } = page;
+  const doc = fs.readFileSync(filename, 'utf8');
+  tryParseImports(doc, filename, platform, cb);
+}
+
+async function tryParseImports(source, filename, platform, cb) {
+  console.log('Compiling', filename);
+  try {
+    const compiled = String(await compile(source));
+    const lines = source.split('\n');
+    const imports = [...(await parseImports(compiled))];
+    const fragments = {};
+    lines.forEach((line, lineNumber) => {
+      if (line.includes('<Fragments')) {
+        allFilters.forEach((filter) => {
+          if (line.includes(filter)) {
+            if (!(filter in fragments)) {
+              fragments[filter] = [];
+            }
+            imports.forEach((parsedImport) => {
+              const isAbsolute = parsedImport.moduleSpecifier.isConstant;
+              const hasDefault = parsedImport.importClause.default;
+              const likelyFragment = isAbsolute && hasDefault;
+              if (
+                likelyFragment &&
+                parsedImport.importClause.default.includes(filter)
+              ) {
+                fragments[filter].push(parsedImport.moduleSpecifier.value);
+              }
+            });
+          }
+        });
+      }
+    });
+
+    // derive URL path
+    filename = filename.split('[');
+    filename[1] = platform;
```

Comment:
Isn't this going to mean that `platform` will look like `framework].mdx` or something?  not sure

Reply:
or the opposite I mean

Reply:
oh okay nvm